### PR TITLE
Add the  internalmetrics file

### PR DESCRIPTION
**Pull request by shavidissa**
The internal metrics file is common to all the Java SDKs too. Added the content in the Wavefront Java sender SDK to this repo.

### CODE REVIEW

File: java/internalmetrics.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+If you are sending data to Wavefront via the Wavefront proxy using a `WavefrontProxyClient` instance, the following diagnostic metrics are reported:
+
+* [Metric Points Handler](#Metric-Points-Handler)
+* [Histogram Distributions Handler](#Histogram-Distributions-Handler)
+* [Tracing Handler](#Tracing-Handler)
+
+### Metric Points Handler
+| Metric Name|Metric Type|Description|
+|:---|:---:|:---|
+|~sdk.java.core.sender.proxy.points.discarded.count                     |Counter    |Points discarded due to port(s) that were not configured.|
```
Comment (rkempf42) [inline java/internalmetrics.md]: Points discarded because port(s) were not configured

File: java/internalmetrics.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+|~sdk.java.core.sender.proxy.points.valid.count                         |Counter    |Valid points received.|
+|~sdk.java.core.sender.proxy.points.invalid.count                       |Counter    |Invalid points received.|
+|~sdk.java.core.sender.proxy.points.dropped.count                       |Counter    |Points dropped due to failure to write to socket.|
```
Comment (rkempf42) [inline java/internalmetrics.md]: Points dropped due to failure to write to socket. => What does that mean?

File: java/internalmetrics.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+|~sdk.java.core.sender.proxy.metricHandler.socket.write.errors.count    |Counter    |Unsuccessful writes to socket.|
+|~sdk.java.core.sender.proxy.metricHandler.socket.flush.success.count   |Counter    |Successful socket flushes.|
+|~sdk.java.core.sender.proxy.metricHandler.socket.flush.errors.count    |Counter    |Unsuccessful socket flushes.|
+|~sdk.java.core.sender.proxy.metricHandler.socket.reset.success.count   |Counter    |Successful connection resets.|
+|~sdk.java.core.sender.proxy.metricHandler.socket.reset.errors.count    |Counter    |Unsuccessful connection resets.|
+
+### Histogram Distributions Handler
+|Metric Name|Metric Type|Description|
+|:---|:---:|:---|
+|~sdk.java.core.sender.proxy.histograms.discarded.count                     |Counter    |Distributions discarded due to port(s) that were not configured.|
```
Comment (rkempf42) [inline java/internalmetrics.md]: (see comment above)

File: java/internalmetrics.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+|~sdk.java.core.sender.proxy.histogramHandler.socket.write.errors.count     |Counter    |Unsuccessful writes to socket.|
+|~sdk.java.core.sender.proxy.histogramHandler.socket.flush.success.count    |Counter    |Successful socket flushes.|
+|~sdk.java.core.sender.proxy.histogramHandler.socket.flush.errors.count     |Counter    |Unsuccessful socket flushes.|
+|~sdk.java.core.sender.proxy.histogramHandler.socket.reset.success.count    |Counter    |Successful connection resets.|
+|~sdk.java.core.sender.proxy.histogramHandler.socket.reset.errors.count     |Counter    |Unsuccessful connection resets.|
+
+### Tracing Handler
+|Metric Name|Metric Type|Description|
+|:---|:---:|:---|
+|~sdk.java.core.sender.proxy.spans.discarded.count                      |Counter    |Spans discarded due to port(s) that were not configured.|
```
Comment (rkempf42) [inline java/internalmetrics.md]: see above

File: java/internalmetrics.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+|~sdk.java.core.sender.proxy.tracingHandler.errors.count                |Counter    |Errors encountered, excluding connection errors. |
+|~sdk.java.core.sender.proxy.tracingHandler.connect.errors.count        |Counter    |Errors encountered connecting to remote socket.|
+|~sdk.java.core.sender.proxy.tracingHandler.socket.write.success.count  |Counter    |Successful writes to socket.|
+|~sdk.java.core.sender.proxy.tracingHandler.socket.write.errors.count   |Counter    |Unsuccessful writes to socket.|
+|~sdk.java.core.sender.proxy.tracingHandler.socket.flush.success.count  |Counter    |Successful socket flushes.|
+|~sdk.java.core.sender.proxy.tracingHandler.socket.flush.errors.count   |Counter    |Unsuccessful socket flushes.|
+|~sdk.java.core.sender.proxy.tracingHandler.socket.reset.success.count  |Counter    |Successful connection resets.|
+|~sdk.java.core.sender.proxy.tracingHandler.socket.reset.errors.count   |Counter    |Unsuccessful connection resets.|
+
+## Metrics Collected When Using Direct Ingestion
```
Comment (rkempf42) [inline java/internalmetrics.md:68]: I would start this with a sort-of disclaimer, like "During Proof-of_Concept implementations, some developers send message directly to the Wavefront service." 
And, start the proxy part with the opposite disclaimer.

File: java/internalmetrics.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+|~sdk.java.core.sender.direct.points.valid.count                |Counter    |Valid points received.|
+|~sdk.java.core.sender.direct.points.invalid.count              |Counter    |Invalid points received.|
+|~sdk.java.core.sender.direct.points.dropped.count              |Counter    |Valid points that are dropped during ingestion.|
+|~sdk.java.core.sender.direct.points.report.errors.count        |Counter    |Exceptions encountered while reporting points.|
+|~sdk.java.core.sender.direct.points.report.202.count           |Counter    |Report requests by response status code.|
+
+### Histogram Distributions Ingestion
+|Metric Name|Metric Type|Description|
+|:---|:---:|:---|
+|~sdk.java.core.sender.direct.histograms.queue.size                 |Gauge      |Distributions in ingestion buffer.|
```
Comment (rkempf42) [inline java/internalmetrics.md:89]: I'm puzzled that isn't a counter, but maybe I'm misunderstanding something.

**Review by rkempf42 — CHANGES_REQUESTED** [review]
Some questions -- if it's not easy to get the answers, OK to push.

File: java/internalmetrics.md (hunk 6 above)
Comment (shavidissa) [reply java/internalmetrics.md:89]: Checked with Han, and he confirmed the current content.

File: java/internalmetrics.md (hunk 5 above)
Comment (shavidissa) [reply java/internalmetrics.md:68]: Will move back to address this later on, as discussed.

**Review by rkempf42 — APPROVED** [review]
LGtM!

**Review by sushantdewan123 — APPROVED** [review]

**Pull request merged**